app/controllers/idv/capture_doc_controller.rb

```diff
@@ -12,15 +12,53 @@ class CaptureDocController < ApplicationController
       analytics_id: Analytics::CAPTURE_DOC,
     }.freeze
 
+    def index
+      ##
+      # This allows us to switch between the old doc auth step by step flow and
+      # the new document capture flow. When the old step by step flow is retired
+      # we should remove this logic and redirect the user directly to the show
+      # action.
+      #
+      if FeatureManagement.document_capture_step_enabled?
+        @flow.flow_session['Idv::Steps::MobileFrontImageStep'] = true
+        @flow.flow_session['Idv::Steps::CaptureMobileBackImageStep'] = true
+        @flow.flow_session['Idv::Steps::SelfieStep'] = true
+      else
+        @flow.flow_session['Idv::Steps::DocumentCaptureStep'] = true
+      end
+      redirect_to_step(next_step)
+    end
+
     private
 
     def ensure_user_id_in_session
-      return if session[:doc_capture_user_id] && token.blank?
-      result = CaptureDoc::ValidateRequestToken.new(token).call
+      return if session[:doc_capture_user_id] &&
+                token.blank? &&
+                document_capture_session_uuid.blank?
+
+      result = if FeatureManagement.document_capture_step_enabled?
+                 validate_document_capture_session_from_uuid
+               else
+                 CaptureDoc::ValidateRequestToken.new(token).call
+               end
+
       analytics.track_event(FSM_SETTINGS[:analytics_id], result.to_h)
       process_result(result)
     end
 
+    def validate_document_capture_session_from_uuid
+      document_capture_session = DocumentCaptureSession.find_by(uuid: document_capture_session_uuid)
+      FormResponse.new(
+        success: document_capture_session.present? && !document_capture_session.expired?,
+        errors: {},
+        extra: {
+          for_user_id: document_capture_session&.user_id,
+          user_id: 'anonymous-uuid',
+          event: 'Request token validation',
```

Comment:
what is this used for? is this supposed to roll to to our analytics log? Typically event names come from a constant in `analytics.rb`

Reply:
> what is this used for? is this supposed to roll to to our analytics log? Typically event names come from a constant in `analytics.rb`

Much of this new logic recreates (duplicates) existing behaviors for the `doc_captures` table and associated classes. This in particular is from:

https://github.com/18F/identity-idp/blob/10cb94f32ae0dc0cfb07d93ffc8200565be5f49b/app/services/capture_doc/validate_request_token.rb#L20-L26

To your point, following the logic I think it's still the case we're using a constant for the event name being tracked for analytics, and that the string here is part of the serialized payload.

https://github.com/18F/identity-idp/blob/10cb94f32ae0dc0cfb07d93ffc8200565be5f49b/app/controllers/idv/capture_doc_controller.rb#L20

https://github.com/18F/identity-idp/blob/10cb94f32ae0dc0cfb07d93ffc8200565be5f49b/app/controllers/idv/capture_doc_controller.rb#L12

Trying to recall if @jmhooper and I had discussed this during our pairing, I'm wondering if we'd want to go down the path of trying to extract something equivalent to `CaptureDoc::ValidateRequestToken` for `DocumentCaptureSession`. Or at least some other refactoring to make it clearer that these are extra analytics attributes (like with [`extra_analytics_attributes`](https://github.com/18F/identity-idp/blob/10cb94f32ae0dc0cfb07d93ffc8200565be5f49b/app/services/capture_doc/validate_request_token.rb#L20)).

Reply:
I refactored this in a15eb77 to be more like how validation was done previously with `CaptureDoc::ValidateRequestToken`.

One difference: For the error messages, I wasn't sure if it made sense to localize the strings. This is what we'd done previously for token errors, but as far as I know, we don't show these errors to users, and the only place they would seem to be relevant is in the analytics, where I would think we'd want consistent texts regardless of locale for better filtering. I could add new strings or just reuse the existing token validation strings if it would be good to do so.

**`RequestTokenValidator`**

https://github.com/18F/identity-idp/blob/05bdedc7bbae3934ed7577a7c1d4b143e07d0a88/app/validators/idv/request_token_validator.rb#L6

https://github.com/18F/identity-idp/blob/05bdedc7bbae3934ed7577a7c1d4b143e07d0a88/app/validators/idv/request_token_validator.rb#L18

https://github.com/18F/identity-idp/blob/05bdedc7bbae3934ed7577a7c1d4b143e07d0a88/app/validators/idv/request_token_validator.rb#L23

**`DocumentCaptureSessionValidator`**

https://github.com/18F/identity-idp/blob/a15eb77bcad55d34d580623f3f8517cd61536da1/app/validators/idv/document_capture_session_validator.rb#L6

https://github.com/18F/identity-idp/blob/a15eb77bcad55d34d580623f3f8517cd61536da1/app/validators/idv/document_capture_session_validator.rb#L17

https://github.com/18F/identity-idp/blob/a15eb77bcad55d34d580623f3f8517cd61536da1/app/validators/idv/document_capture_session_validator.rb#L22